<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QWidget" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1027</width>
    <height>830</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget" native="true">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>900</width>
     <height>530</height>
    </rect>
   </property>
   <widget class="QPushButton" name="loadButton">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>0</y>
      <width>80</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Загрузить</string>
    </property>
   </widget>
   <widget class="scenezoom" name="originalImage">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>180</y>
      <width>481</width>
      <height>421</height>
     </rect>
    </property>
    <property name="verticalScrollBarPolicy">
     <enum>Qt::ScrollBarAsNeeded</enum>
    </property>
    <property name="sizeAdjustPolicy">
     <enum>QAbstractScrollArea::AdjustToContents</enum>
    </property>
    <property name="transformationAnchor">
     <enum>QGraphicsView::AnchorUnderMouse</enum>
    </property>
    <property name="resizeAnchor">
     <enum>QGraphicsView::AnchorUnderMouse</enum>
    </property>
   </widget>
   <widget class="scenezoom" name="filteredScene">
    <property name="geometry">
     <rect>
      <x>510</x>
      <y>180</y>
      <width>501</width>
      <height>421</height>
     </rect>
    </property>
    <property name="transformationAnchor">
     <enum>QGraphicsView::AnchorUnderMouse</enum>
    </property>
    <property name="resizeAnchor">
     <enum>QGraphicsView::AnchorUnderMouse</enum>
    </property>
   </widget>
   <widget class="QPushButton" name="sobelButton">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>0</y>
      <width>80</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Фильтр</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="kernelSize">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>0</y>
      <width>111</width>
      <height>26</height>
     </rect>
    </property>
    <property name="minimum">
     <number>3</number>
    </property>
    <property name="maximum">
     <number>9</number>
    </property>
    <property name="singleStep">
     <number>2</number>
    </property>
   </widget>
   <widget class="QLineEdit" name="kernelSizeTitle">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>0</y>
      <width>113</width>
      <height>25</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="autoFillBackground">
     <bool>true</bool>
    </property>
    <property name="text">
     <string>Размер ядра</string>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QSlider" name="xOrderSlider">
    <property name="geometry">
     <rect>
      <x>530</x>
      <y>0</y>
      <width>160</width>
      <height>16</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="maximum">
     <number>1</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QSlider" name="yOrderSlider">
    <property name="geometry">
     <rect>
      <x>530</x>
      <y>40</y>
      <width>160</width>
      <height>16</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="maximum">
     <number>1</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="zeroXOrderLabel">
    <property name="geometry">
     <rect>
      <x>530</x>
      <y>20</y>
      <width>16</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>0</string>
    </property>
   </widget>
   <widget class="QLabel" name="oneXOrderLabel">
    <property name="geometry">
     <rect>
      <x>680</x>
      <y>20</y>
      <width>16</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>1</string>
    </property>
   </widget>
   <widget class="QLabel" name="zeroYOrderLabel">
    <property name="geometry">
     <rect>
      <x>530</x>
      <y>60</y>
      <width>16</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>0</string>
    </property>
   </widget>
   <widget class="QLabel" name="oneYOrderLabel">
    <property name="geometry">
     <rect>
      <x>680</x>
      <y>50</y>
      <width>20</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>1</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>0</y>
      <width>80</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Фильтр</string>
    </property>
   </widget>
   <widget class="QLabel" name="xLabel">
    <property name="geometry">
     <rect>
      <x>510</x>
      <y>0</y>
      <width>16</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>x</string>
    </property>
   </widget>
   <widget class="QLabel" name="yLabel">
    <property name="geometry">
     <rect>
      <x>510</x>
      <y>40</y>
      <width>20</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>y</string>
    </property>
   </widget>
   <widget class="QPushButton" name="saveButton">
    <property name="geometry">
     <rect>
      <x>730</x>
      <y>0</y>
      <width>80</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Сохранить</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_2">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>0</y>
      <width>80</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Бинаризация</string>
    </property>
   </widget>
   <widget class="QSlider" name="thresholdTypeSlider">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>30</y>
      <width>261</width>
      <height>16</height>
     </rect>
    </property>
    <property name="maximum">
     <number>4</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="thresholdTypeLabel">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>50</y>
      <width>401</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>0:Обычная;1:Инверсная;2:Пороговая;3:К нулю;4:Инверсия к нулю</string>
    </property>
   </widget>
   <widget class="QScrollBar" name="thresholdValue">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>50</y>
      <width>351</width>
      <height>16</height>
     </rect>
    </property>
    <property name="maximum">
     <number>255</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLineEdit" name="thresholdValueDisplay">
    <property name="geometry">
     <rect>
      <x>830</x>
      <y>20</y>
      <width>41</width>
      <height>25</height>
     </rect>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="dilatationButton">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>0</y>
      <width>80</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Дилатация</string>
    </property>
   </widget>
   <widget class="QSlider" name="dilatationTypeSlider">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>70</y>
      <width>261</width>
      <height>16</height>
     </rect>
    </property>
    <property name="maximum">
     <number>2</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QSpinBox" name="dilatationKernelSizeSpinBox">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>60</y>
      <width>45</width>
      <height>26</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>21</number>
    </property>
    <property name="value">
     <number>1</number>
    </property>
   </widget>
   <widget class="QLabel" name="dilatationTypeLabel">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>90</y>
      <width>261</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string> 1. Квадрат                         2.Крест              3.Круг</string>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>90</y>
      <width>101</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Размер ядра</string>
    </property>
   </widget>
   <widget class="QPushButton" name="prewittButton">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>0</y>
      <width>80</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Фильтр</string>
    </property>
   </widget>
   <widget class="QPushButton" name="medianButton">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>0</y>
      <width>80</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Фильтр</string>
    </property>
   </widget>
   <widget class="QPushButton" name="billaterialButton">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>0</y>
      <width>81</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Фильтр</string>
    </property>
   </widget>
   <widget class="QSplitter" name="splitter">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>30</y>
      <width>103</width>
      <height>136</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
    <property name="opaqueResize">
     <bool>false</bool>
    </property>
    <property name="childrenCollapsible">
     <bool>false</bool>
    </property>
    <widget class="QRadioButton" name="radioButton">
     <property name="text">
      <string>Щарр</string>
     </property>
    </widget>
    <widget class="QRadioButton" name="sobelRadioButton">
     <property name="text">
      <string>Собель</string>
     </property>
    </widget>
    <widget class="QRadioButton" name="thresholdButton">
     <property name="text">
      <string>Бинаризация</string>
     </property>
    </widget>
    <widget class="QRadioButton" name="robertsRadioButton">
     <property name="text">
      <string>Робертс</string>
     </property>
    </widget>
    <widget class="QRadioButton" name="dilatationRadioButton">
     <property name="text">
      <string>Дилатация</string>
     </property>
    </widget>
    <widget class="QRadioButton" name="prewittRadioButton">
     <property name="text">
      <string>Превитт</string>
     </property>
    </widget>
    <widget class="QRadioButton" name="billaterialRadioButton">
     <property name="text">
      <string>Билатеральный</string>
     </property>
    </widget>
    <widget class="QRadioButton" name="medianRadioButton">
     <property name="text">
      <string>Медианный</string>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>610</y>
      <width>471</width>
      <height>30</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QPushButton" name="inpaintButton">
       <property name="text">
        <string>Закрасить оригинальное изображение</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="switchSceneButton">
       <property name="text">
        <string>Поменять местами</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QPushButton" name="unhairButton">
    <property name="geometry">
     <rect>
      <x>894</x>
      <y>610</y>
      <width>121</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Убрать волосы</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1027</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>4</width>
     <height>12</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>3</width>
     <height>20</height>
    </rect>
   </property>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>scenezoom</class>
   <extends>QGraphicsView</extends>
   <header location="global">ilya/scenezoom.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
